lib/commands/build.js

```diff
@@ -0,0 +1,56 @@
+'use strict'
+const Build = module.exports = {
+  command: 'build',
+  describe: 'Executes the configured build script, if present, or executes ' +
+    ' silently',
+  builder (y) {
+    return y.help().alias('help', 'h')
+      .options(Build.options)
+  },
+  options: Object.assign(require('../common-opts.js', {})),
+  handler: async argv => build(argv)
+}
+
+async function build (argv) {
+  const findPrefix = require('find-npm-prefix')
+  const readJson = require('read-package-json')
+  const figgyPudding = require('figgy-pudding')
+  const npmConfig = require('../config.js')
+  const libnpm = require('libnpm')
+  const log = require('npmlog')
+
+  const cwd = process.cwd()
+  const BuildConfig = figgyPudding({
+    log: { default: () => log },
+    dir: { default: cwd },
+    production: { default: false },
+    nodeOptions: { default: {} },
+    config: { default: {} },
+    unsafePerm: { default: false },
+    scriptsPrependNodePath: { default: false },
+    ignoreScripts: { default: false },
+    user: {},
+    group: {},
+    stdio: {},
+    scriptShell: {}
+  })
```

Comment:
Minor FYI: these default values of the `BuildConfig` are based on my best guesses, while reading code about how these option keys are being used in [npm-lifecycle/index.js](https://github.com/npm/npm-lifecycle/blob/latest/index.js).

Please advise if any of these default values, or the way of how this config object is written, can be improved. 🙏